src/compiler/scala/tools/nsc/settings/ScalaSettings.scala

```diff
@@ -461,7 +458,7 @@ trait ScalaSettings extends AbsScalaSettings
 
   /** Groups of Settings.
    */
-  val future        = BooleanSetting("-Xfuture", "Turn on future language features.") enablingIfNotSetByUser futureSettings
+  val future        = BooleanSetting("-Xfuture", "Deprecated. Does nothing.").withDeprecationMessage("Use -Xsource instead.")
```

Comment:
There's no future
In England's dreaming

Reply:
I came here to say "there is no future", but instead I am grateful for @eed3si9n . Note that maybe it's worse if the future "does nothing".

Reply:
Updated message. It needn't be gloomy.